include/swift/SIL/SILBuilder.h

```diff
@@ -40,6 +41,40 @@ class SILBuilder {
   /// recorded in this list.
   SmallVectorImpl<SILInstruction *> *InsertedInstrs = nullptr;
 
+  /// Maps opened archetypes to their definitions. If provided,
+  /// can be used by the builder. It is supposed to be used
+  /// only by SILGen or SIL deserializers.
+  SILOpenedArchetypesTracker *OpenedArchetypes = nullptr;
+
+  bool canProvideTypeDefs() const {
+    return OpenedArchetypes != nullptr;
+  }
+
+  // Build a set of typedef operands for an apply instruction. This is done by
+  // scanning the list of substitutions and collecting all opened archetypes.
+  void createApplyTypeDefs(SILInstruction *I, ArrayRef<Substitution> Subs) {
```

Comment:
These are opened archetype operands that refer to opened existential values. We're not even creating special "def" values for archetypes yet, so let's not call them "defs". OpenedArchetype is fine.